kube/registry_credentials.go

```diff
@@ -37,6 +37,9 @@ func MakeRegistryCredentials(settings ExportSettings) (helm.Node, error) {
 		SetAPIVersion("v1").
 		SetKind("Secret").
 		SetName("registry-credentials")
+	if settings.CreateHelmChart {
+		cb.AddModifier(helm.Block(`if ne .Values.kube.registry.username ""`))
```

Comment:
The actual problem we are dealing with is docker hub, so we are good for that one.

I don't think it makes sense to have a password without a username, but it would be easy enough to change the condition to

```
if or (ne .Values.kube.registry.username "") (ne .Values.kube.registry.password "")
```

I don't know why passwordless usernames would be a problem for us though. You can specify them using the template, but if the backend doesn't support them, then you wouldn't specify them anyways, so either way the templates work as long as the information provided is correct.

Reply:
Good point re: matching the backend, so I think we're good. (Though the people pushing the images (us) may not be the people running the Kube deployments… but if we only have one they're hosed regardless of what this bit does)